opensrp-pnc/src/main/java/org/smartregister/pnc/repository/PncChildRepository.java

```diff
@@ -94,19 +107,75 @@ public boolean delete(PncChild pncChild) {
         throw new NotImplementedException("");
     }
 
-    public int countBaby28DaysOld(String baseEntityId, int howBabyOldInDays) {
+    public List<PncChild> findAll(@NonNull String medicInfoId) {
+        List<PncChild> data = new ArrayList<>();
+        SQLiteDatabase sqLiteDatabase = getReadableDatabase();
+        try (Cursor cursor = sqLiteDatabase.rawQuery("SELECT * FROM " + PncDbConstants.Table.PNC_BABY + " WHERE " + PncDbConstants.Column.PncBaby.MEDIC_INFO_ID + "='" + medicInfoId + "'", null)) {
+            while (cursor.moveToNext()) {
+                data.add(fillUp(cursor));
+            }
+        }
+
+        return data;
+    }
+
+    public List<HashMap<String, String>> findAllByBaseEntityId(@NonNull String baseEntityId) {
```

Comment:
can this method name be more specific ... and is this method necessary?

Reply:
method renamed.